bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,529 @@
+                                    updateState(CHANNEL_TYPE_FANSPEED, StringType.valueOf("AUTO"));
+                                } else {
+                                    updateState(CHANNEL_TYPE_FANSPEED,
+                                            StringType.valueOf(String.valueOf(element.value)));
+                                }
+                                break;
+                            case 5:
+                            case 6:
+                                State state;
+                                if ((element.value) == 0) {
+                                    state = StringType.valueOf("AUTO");
+                                } else if ((element.value) == 10) {
+                                    state = StringType.valueOf("SWING");
+                                } else if ((element.value) == 11) {
+                                    state = StringType.valueOf("SWIRL");
+                                } else if ((element.value) == 12) {
+                                    state = StringType.valueOf("WIDE");
+                                } else {
+                                    state = StringType.valueOf(String.valueOf(element.value));
+                                }
+                                switch (element.uid) {
+                                    case 5:
+                                        updateState(CHANNEL_TYPE_VANESUD, state);
+                                        break;
+                                    case 6:
+                                        updateState(CHANNEL_TYPE_VANESLR, state);
+                                        break;
+                                }
+                                break;
+                            case 9:
+                                int unit = Math.round((element.value) / 10);
+                                State stateValue = QuantityType.valueOf(unit, SIUnits.CELSIUS);
+                                updateState(CHANNEL_TYPE_TARGETTEMP, stateValue);
+                                break;
+                            case 10:
+                                unit = Math.round((element.value) / 10);
+                                stateValue = QuantityType.valueOf(unit, SIUnits.CELSIUS);
+                                updateState(CHANNEL_TYPE_AMBIENTTEMP, stateValue);
+                                break;
+                            case 37:
+                                unit = Math.round((element.value) / 10);
+                                stateValue = QuantityType.valueOf(unit, SIUnits.CELSIUS);
+                                updateState(CHANNEL_TYPE_OUTDOORTEMP, stateValue);
+                                break;
+                        }
+                    }
+                }
+            } catch (JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        }
+    }
+
+    public void login() {
```

Comment:
Since `login()` can fail, you should provide a way for callers to know if it executed without issue. This can be done by either adding a return value that indicates success or by throwing an exception if something goes wrong.

Since you are currently having this method update the thing status it would be easiest to just have it return the sessionId.

```suggestion
    public String login() {
```

Then it could be called like this:
```
String sessionId = login();
if(sessionId != null){
...
    logout(sessionId);
}
```

If you want to take things further, a better way would be to refactor session management into a single method and allow callers to pass the functions that they want executed in a single session.
Lambdas make this easy.
```
private void postRequest(UnaryOperator<String> requestFactory, Consumer<Response> handler){
  String sessionId = login();
  if(sessionId != null){
    try{
      String request = requestFactory.apply(sessionId);
      String response = api.postRequest(ipAddress, contentString);
      if (response != null) {
            try {
                Response resp = gson.fromJson(response, Response.class);
                boolean success = resp.success;
                if (success) {
                   handler.accept(resp);
                } else {
                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Request unsuccessful");
                }
      }else{
         updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "No Response");
      }
    } catch (JsonSyntaxException e) {
      updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
    } finally {
      logout(sessionId);
    }
  }
}
```

Then it could be called like this:
```

private void handleDataPointsResponse(Response response){
}

private void foo(){
    postRequest(sessionId -> 
       "{\"command\":\"getavailabledatapoints\",\"data\":{\"sessionID\":\"" + sessionId + "\"}}", 
        this::handleDataPointsResponse);
}
```

Reply:
I have changed the sessionID handling as requested, but would like to leve your further refactoring for a later stage.
A fist try to implement this, which indeed seems to be a good idea, lead into some errors.

Reply:
Implemented this as well